### PR TITLE
feat(evm): create rpc method for namespace registration

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

The goal here is to extend the EVM base shard to have a `Register` method for game shards to register themselves to this evm base shard. The intent is that when `Cardinal` runs `StartGame` it will send a `Register` message to the EVM base shard, which will then save it's `namespace` : `router_address` pair. This will then allow any `Router` callers to simply tell the `Router` the `namespace` of the game shard they want to interact with, and the base shard will get the address for that namespace under the hood.

## Summary by CodeRabbit

- **New Features**
	- Extended functionality to register game shard requests and process transactions before sequencing.
	- Introduced a new RPC method for registering game shards in the Router system.
- **Bug Fixes**
	- Updated the handling of namespace addresses to use dynamic values instead of hardcoded ones.
- **Refactor**
	- Removed unused imports and environment variable dependencies to streamline configuration.
	- Renamed and modified methods in the transaction queue for clarity and improved functionality.
- **Tests**
	- Adjusted test cases to align with the new return types and methods introduced in the sequencer server and transaction queue.

### CODE REVIEW

**Comment by technicallyty** [comment]
@coderabbitai review

File: evm/x/namespace/keeper/keeper.go [hunk 1]
```diff
@@ -14,6 +18,12 @@ type Keeper struct {
 }
 
 func NewKeeper(storeKey *storetypes.KVStoreKey, auth string) *Keeper {
+	if auth == "" {
+		auth = authtypes.NewModuleAddress(sequencer.Name).String()
+		if strings.Contains(auth, "cosmos") {
+			log.Fatal("address had 'cosmos' bech32 prefix, should be 'world'")
```
Comment (jerargus) [inline evm/x/namespace/keeper/keeper.go:24]: Should this check be fore a "cosmos" prefix, or just that it contains cosmos? The error message and if check are not in agreement.

Comment (technicallyty) [reply evm/x/namespace/keeper/keeper.go:24]: [Contains -> HasPrefix](https://github.com/Argus-Labs/world-engine/pull/626/commits/e35886b7bd703ab3c5375a118a78d22e59651f1f)
fixed

**Review by pyrofolium — APPROVED** [review]
lgtm

**Pull request merged**